Super flakey a4a integration test

```
Chrome 54.0.2840 (Linux 0.0.0) integration test: a4a should render a single AMP ad in a friendly iframe FAILED
	AssertionError: iframe doc: expected null to be truthy
	    at doAsserterAsyncAndAddThen (/home/travis/build/ampproject/amphtml/node_modules/chai-as-promised/lib/chai-as-promised.js:292:33)
	    at Assertion.<anonymous> (/home/travis/build/ampproject/amphtml/node_modules/chai-as-promised/lib/chai-as-promised.js:282:25)
	    at Assertion.get (/home/travis/build/ampproject/amphtml/node_modules/chai/chai.js:5396:37)
	    at expectRenderedInFriendlyIframe (/tmp/1b36fa6b09ef035ee51fbc50e2dcc2b8.browserify:11583:44 <- /home/travis/build/ampproject/amphtml/extensions/amp-a4a/0.1/test/test-a4a-integration.js:49:44)
	    at /tmp/1b36fa6b09ef035ee51fbc50e2dcc2b8.browserify:11653:7 <- /home/travis/build/ampproject/amphtml/extensions/amp-a4a/0.1/test/test-a4a-integration.js:118:6
```

--- a/extensions/amp-a4a/0.1/test/test-a4a-integration.js
+++ b/extensions/amp-a4a/0.1/test/test-a4a-integration.js
@@ -50,7 +50,7 @@ import * as sinon from 'sinon';
  * @param {!Element} element amp-ad element to examine.
  * @param {string} srcdoc  A string that must occur somewhere in the friendly
  *   iframe `srcdoc` attribute.
- * @return {!Promise<boolean>} Promise that executes assertions on friendly
+ * @return {!Promise} Promise that executes assertions on friendly
  *   iframe contents.
  */
 function expectRenderedInFriendlyIframe(element, srcdoc) {
@@ -64,7 +64,6 @@ function expectRenderedInFriendlyIframe(element, srcdoc) {
     expect(element, 'ad tag').to.be.visible;
     expect(child, 'iframe child').to.be.visible;
     expect(childDocument, 'ad creative content doc').to.be.visible;
-    return true;
   });
 }
 
@@ -81,7 +80,7 @@ function expectRenderedInXDomainIframe(element, src) {
   expect(child, 'iframe child').to.be.visible;
 }
 
-describe.only('integration test: a4a', () => {
+describe('integration test: a4a', () => {
   let sandbox;
   let xhrMock;
   let fixture;

--- a/extensions/amp-a4a/0.1/test/test-a4a-integration.js
+++ b/extensions/amp-a4a/0.1/test/test-a4a-integration.js
@@ -55,8 +55,6 @@ import * as sinon from 'sinon';
  */
 function expectRenderedInFriendlyIframe(element, srcdoc) {
   expect(element, 'ad element').to.be.ok;
-  expect(element.querySelectorAll('iframe'),
-    'amp-ad tag should have single iframe child').to.have.lengthOf(1);
   const child = element.querySelector('iframe[srcdoc]');
   expect(child, 'iframe child').to.be.ok;
   expect(child.getAttribute('srcdoc')).to.contain.string(srcdoc);
@@ -74,8 +72,6 @@ function expectRenderedInXDomainIframe(element, src) {
   // Note: Unlike expectRenderedInXDomainIframe, this doesn't return a Promise
   // because it doesn't (cannot) inspect the contents of the iframe.
   expect(element, 'ad element').to.be.ok;
-  expect(element.querySelectorAll('iframe'),
-      'amp-ad tag should have single iframe child').to.have.lengthOf(1);
   expect(element.querySelector('iframe[srcdoc]'),
       'does not have a friendly iframe child').to.not.be.ok;
   const child = element.querySelector('iframe[src]');

--- a/extensions/amp-a4a/0.1/test/test-a4a-integration.js
+++ b/extensions/amp-a4a/0.1/test/test-a4a-integration.js
@@ -34,6 +34,7 @@ import {
 } from '../../../../src/custom-element';
 import {utf8Encode} from '../../../../src/utils/bytes';
 import '../../../amp-ad/0.1/amp-ad-xorigin-iframe-handler';
+import {loadPromise} from '../../../../src/event-helper';
 import * as sinon from 'sinon';
 
 // Integration tests for A4A.  These stub out accesses to the outside world
@@ -41,20 +42,40 @@ import * as sinon from 'sinon';
 // otherwise test the complete A4A flow, without making assumptions about
 // the structure of that flow.
 
+/**
+ * Checks various consistency properties on the friendly iframe created by
+ * A4A privileged path rendering.  Note that this returns a Promise, so its
+ * value must be returned from any test invoking it.
+ *
+ * @param {!Element} element amp-ad element to examine.
+ * @param {string} srcdoc  A string that must occur somewhere in the friendly
+ *   iframe `srcdoc` attribute.
+ * @return {!Promise<boolean>} Promise that executes assertions on friendly
+ *   iframe contents.
+ */
 function expectRenderedInFriendlyIframe(element, srcdoc) {
   expect(element, 'ad element').to.be.ok;
+  expect(element.querySelectorAll('iframe'),
+    'amp-ad tag should have single iframe child').to.have.lengthOf(1);
   const child = element.querySelector('iframe[srcdoc]');
   expect(child, 'iframe child').to.be.ok;
   expect(child.getAttribute('srcdoc')).to.contain.string(srcdoc);
-  const childDocument = child.contentDocument.documentElement;
-  expect(childDocument, 'iframe doc').to.be.ok;
-  expect(element, 'ad tag').to.be.visible;
-  expect(child, 'iframe child').to.be.visible;
-  expect(childDocument, 'ad creative content doc').to.be.visible;
+  return loadPromise(child).then(() => {
+    const childDocument = child.contentDocument.documentElement;
+    expect(childDocument, 'iframe doc').to.be.ok;
+    expect(element, 'ad tag').to.be.visible;
+    expect(child, 'iframe child').to.be.visible;
+    expect(childDocument, 'ad creative content doc').to.be.visible;
+    return true;
+  });
 }
 
 function expectRenderedInXDomainIframe(element, src) {
+  // Note: Unlike expectRenderedInXDomainIframe, this doesn't return a Promise
+  // because it doesn't (cannot) inspect the contents of the iframe.
   expect(element, 'ad element').to.be.ok;
+  expect(element.querySelectorAll('iframe'),
+      'amp-ad tag should have single iframe child').to.have.lengthOf(1);
   expect(element.querySelector('iframe[srcdoc]'),
       'does not have a friendly iframe child').to.not.be.ok;
   const child = element.querySelector('iframe[src]');
@@ -64,7 +85,7 @@ function expectRenderedInXDomainIframe(element, src) {
   expect(child, 'iframe child').to.be.visible;
 }
 
-describe('integration test: a4a', () => {
+describe.only('integration test: a4a', () => {
   let sandbox;
   let xhrMock;
   let fixture;
@@ -116,7 +137,7 @@ describe('integration test: a4a', () => {
 
   it('should render a single AMP ad in a friendly iframe', () => {
     return fixture.addElement(a4aElement).then(unusedElement => {
-      expectRenderedInFriendlyIframe(a4aElement, 'Hello, world.');
+      return expectRenderedInFriendlyIframe(a4aElement, 'Hello, world.');
     });
   });
 

